specs/WinRT/MakeMSIX-spec.md

```diff
@@ -0,0 +1,959 @@
+        platformSDKExecutablePath /= L"bin";
+        platformSDKExecutablePath /= productVersionString;
+        platformSDKExecutablePath /= processorArchitecture;
+
+        return platformSDKExecutablePath;
+    }
+
+    void LaunchSignToolWithArguments(std::wstring arguments)
+    {
+        std::filesystem::path signToolPath = GetPlatformSDKPath();
+        signToolPath /= L"signtool.exe";
+
+        std::wstring commandLine{ signToolPath };
+        commandLine += L" " + arguments;
+        DWORD exitCode{};
+        winrt::check_hresult(CreateProcessAndWaitForExitCode(commandLine, exitCode));
+        if (exitCode == EXIT_FAILURE)
+        {
+            winrt::throw_hresult(E_FAIL);
+        }
+    }
+
+    void Example_SignPackage(std::wstring pfxFile, std::wstring packageFile)
+    {
+        std::wstring signToolArguments;
+        signToolArguments.append(L"sign /fd SHA256 /f ");
+        signToolArguments.append(pfxFile);
+        signToolArguments.append(L" " + packageFile);
+        LaunchSignToolWithArguments(signToolArguments);
+    }
+```
+
+# 5. API Details
+
+```c#
+// Copyright (c) Microsoft Corporation and Contributors.
+// Licensed under the MIT License.
+
+namespace Microsoft.Kozani.MakeMSIX
+{
+    [contractversion(1)]
+    apicontract MakeMSIXContract{};
+
+    /// Options for creating a package
+    [contract(MakeMSIXContract, 1)]
+    runtimeclass PackOptions
+    {
+        /// <summary>
+        /// Created by callers to pass options into CreatePackage
+        /// </summary>
+        PackOptions();
+
+        /// <summary>
+        /// Output path for the full packaged file.
+        /// </summary>
+        String PackageFilePath{ get; set; };
+
+        /// <summary>
+        /// Overwrite PackageFilePath if it already exists.
+        /// Defaults to true.
+        /// </summary>
+        Boolean OverwriteFiles{ get; set; };
+        /// <summary>
+        /// Validates elements of the package during creation.
+        /// Defaults to false.
+        /// </summary>
+        Boolean ValidateFiles{ get; set; };
+    };
+
+    /// Options for creating a Kozani package
+    [contract(MakeMSIXContract, 1)]
+    runtimeclass CreateKozaniPackageOptions
+    {
+        /// <summary>
+        /// 
+        /// </summary>
+        CreateKozaniPackageOptions();
+
+        /// <summary>
+        /// Output path for the kozani packaged file.
+        /// Format must match input file. If packageFilePathToConvert is an appx or msix, then PackageFilePath must be an appx or msix.
+        /// If packageFilePathToConvert is an appxbundle or msixbundle, then PackageFilePath must be an appxbundle or msixbundle.
+        /// </summary>
+        String PackageFilePath{ get; set; };
+
+        /// <summary>
+        /// Optional replacement package publisher.
+        /// If not set, publisher from appxmanifest.xml is unchaged.
+        /// </summary>
+        String PackagePublisher{ get; set; };
```

Comment:
Agreed. Upon implementing the prototype, I realized I do need this packageidentity class because it would be incredibly annoying to have an api that lets you change the publisher name but no api that reads the publisher name from a package. Have added GetPackageIdentity as well.